documentation/guide-service-versioning.asciidoc

```diff
@@ -128,31 +128,31 @@ In general anything can be used to differentiate versions of a service. Possibil
 
 == Practice
-So assuming you know that you have to do service versioning the question is how to do it practically in the code.
-The approach is as following for your devon4j project in case of code-first:
+So assuming you know that you have to do service versioning, the question is how to do it practically in the code.
+The approach for your devon4j project in case of code-first should be as described below:
 
-* Determine which types in the code need to be changed. That is for sure the API and implementation of the according service but most likely also impacts transfer objects and potentially even datatypes.
+* Determine which types in the code need to be changed. It is likely to be the API and implementation of the according service but it may also impact transfer objects and potentially even datatypes.
 * Create new packages for all these concerned types containing the current version number (e.g. `v1`).
 * Copy all these types to that new packages.
 * Rename these copies so they carry the version number as suffix (e.g. `V1`).
 * Increase the version of the service in the unversioned package (e.g. from `v1` to `v2`).
 * Now you have two versions of the same service (e.g. `v1` and `v2`) but so far they behave exactly the same.
 * You start with your actual changes and modify the original files that have been copied before.
 * You will also ensure the links (import statements) of the copied types point to the copies with the version number
-* This will cause incompatibilities (and compile errors) in the copied service. Therefore you need to fix that service implementation to map from the old API to the new API and behavior. In some cases this may be easy (e.g. mapping `x.y.z.v1.FooTo` to `x.y.z.FooTo` using link:guide-beanmapping.asciidoc[bean-mapping] with some custom mapping for the incompatible change), in other cases this can get very complex. Be aware of this complexity from the start before you make your decision about service versioning. 
-* As far as possible this mapping should be done in the service-layer, to not pollute your business code in the core-layer with versioning-aspects. If there is no way to handle it in the service layer, e.g. you need some data from the persistence-layer, implement the "mapping" in the core-layer then, but don't forget to remove this code, when removing the old service version.
-* Finally ensure that both the old service behaves as before as well as the new service works as planned.
+* This will cause incompatibilities (and compile errors) in the copied service. Therefore, you need to fix that service implementation to map from the old API to the new API and behavior. In some casesc, this may be easy (e.g. mapping `x.y.z.v1.FooTo` to `x.y.z.FooTo` using link:guide-beanmapping.asciidoc[bean-mapping] with some custom mapping for the incompatible changes), in other cases this can get very complex. Be aware of this complexity from the start before you make your decision about service versioning.
```

Comment:
`cases` seemed correct here. I guess `casesc` was a type when you wanted to add a comma.